### PR TITLE
feat: add ability to check for multiple default ssh key locations

### PR DESCRIPTION
**Pull request by a-dubs**
Since Ubuntu 24.04 LTS and newer, the default location for ssh keys is
`~/.ssh/id_ed25519` instead of `~/.ssh/id_rsa`. Thus, when no ssh key
path is specified, this commit changes pycloudlib to check for an
existing ssh key at either  `~/.ssh/id_ed25519` or `~/.ssh/id_rsa`.

Fixes GH #341

---

### Testing id_ed25519 can be found
1. made sure no key exists at `~/.ssh/id_rsa`
2. created new key at `~/.ssh/id_ed25519`
3. removed `public_key_path` and `private_key_path` from `[oci]` config in my pycloudlib.toml config file
4. ran the oracle example script: `python3 examples/oracle.py`
5. made sure the following line showed up in the log (was one of the first few lines logged):
```
INFO:pycloudlib.cloud.OCI:using SSH pubkey: /home/a-dubs/.ssh/id_ed25519.pub
```

### Testing error is raised when neither default key is found
1. Removed keys at `~/.ssh/id_ed25519`
2. ran the oracle example script: `python3 examples/oracle.py`
3. Ensured error stating the following was thrown:
```
pycloudlib.errors.PycloudlibError: No public key path provided and no key found in default locations: '~/.ssh/id_rsa.pub' or '~/.ssh/id_ed25519.pub'
```

### Testing error is raised when key from config does not exist:
1. set `public_key_path` in `[oci]` config in my pycloudlib.toml config file to be `~/.ssh/nonexistant-key.pub`  
2. ran the oracle example script: `python3 examples/oracle.py`
3. Ensured error stating the following was thrown:
```
pycloudlib.errors.PycloudlibError: Provided public key path '/home/a-dubs/.ssh/oracle-ipv6-testing.pub1' does not exist
```

### CODE REVIEW

**Comment by a-dubs** [comment]
putting this into draft until i fix tests.

**Comment by a-dubs** [comment]
Also, should i add new unit tests that verify the functionality of the added changes? @holmanb (tag you're it)

File: pycloudlib/cloud.py [hunk 1]
```diff
@@ -62,13 +66,39 @@ def __init__(
         self._check_and_set_config(config_file, required_values)
 
         user = getpass.getuser()
+        # check if id_rsa or id_ed25519 keys exist in the user's .ssh directory
```
Comment (holmanb) [inline pycloudlib/cloud.py:69]: This turns `__init__()` mostly into a key-handling function. I would prefer a helper method `_get_ssh_keys()`[1] to contain this logic and get called from `__init__()`

[1] maybe a better name, I didn't put any thought into that one

**Review by holmanb — COMMENTED** [review]
> Also, should i add new unit tests that verify the functionality of the added changes? @holmanb (tag you're it)

Pycloudlib's test coverage is unfortunately very light on tests, but in this case the feature being tested is something that a user would immediately notice if it isn't working.

I'm honestly not sure whether it would be worth the effort in this case to write and maintain a test given the infrequency of changes to this part of the code and the immediate feedback that we would receive.

What do you think?

File: pycloudlib/cloud.py (hunk 1 above)
Comment (a-dubs) [reply pycloudlib/cloud.py:69]: Haha agreed! Way too much going on there. And would allow for actually unit testing that (if so desired)

**Comment by a-dubs** [comment]
> > Also, should i add new unit tests that verify the functionality of the added changes? @holmanb (tag you're it)
> 
> Pycloudlib's test coverage is unfortunately very light on tests, but in this case the feature being tested is something that a user would immediately notice if it isn't working.
> 
> I'm honestly not sure whether it would be worth the effort in this case to write and maintain a test given the infrequency of changes to this part of the code and the immediate feedback that we would receive.
> 
> What do you think?

Let me go fix the existing unit tests that this change broke, and see if this change gets inherently tested that way. If not, it should be sufficient to just test the helper function that I'll create encapsulating this functionality.

**Comment by a-dubs** [comment]
@holmanb ready for re-review!

File: tests/unit_tests/openstack/test_cloud.py [hunk 2]
```diff
@@ -15,7 +15,7 @@
 
 """
 
-
```
Comment (holmanb) [inline tests/unit_tests/openstack/test_cloud.py:18]: Whitespace

**Review by holmanb — APPROVED** [review]
@a-dubs Nit: I see some unnecessary whitespace changes in diff, but other than that everything else looks good. Please clean up the whitespace changes, then feel free to merge.

**Comment by a-dubs** [comment]
> @a-dubs Nit: I see some unnecessary whitespace changes in diff, but other than that everything else looks good. Please clean up the whitespace changes, then feel free to merge.

@holmanb all those double whitespaces are because they are between classes not functions and this is the formatting enforced by ruff. to verify, I removed the double whitespace then ran `tox -e format` and it added the second newline back in.

**Pull request merged**